### DISCUSSION TITLE
Collection Movement between Projects

### DISCUSSION THREAD

**Discussion by ManuelTS** [Kong/insomnia#5566] [Ideas] [answered]
**Story behind this idea**
I was amazed after the update bringing projects to insomnia. About 30 seconds later I was upset how it was done.

I could not figure out how to easily and quickly move some of my various collections from the default into a new project.

Is collection movement between projects even possible currently?

**Idea**
Make it possible to move collections easily between projects, as many users may have a lot of existing collections in the default Insomnia project after the update and want to sort their set of collections into subsets, i.e. projects.

Make it user friendly and focus on usability by giving multiple ways for moving. The single components needed for that can be reused. The move collection actions can be:

  - Drag 'n Drop (DnD), click and hold the click on a collection in the project view, drag it to a project, release the mouse button. Result is that the collection is added to the project which was below the cursor during the mouse button release. Otherwise don't execute the movement
  - Add a "Move" entry to the context menu of a collection in the project view, a click on the collection context move action entry opens a dialog displaying all projects and the user can select all projects to move to except the one the collection is already in
  - Same for the duplicate entry, let the user decide to which target project, in this case, it can be the current project too
  - Add to the "Export" that one can export a collection from one project to another one
  - Add to the "Import" that one can import a collection from one project to another one 

**Insomnia Version**
2022.7.0

**Comment by ManuelTS** [comment]
Still nothing and so few upvotes? Sad, very sad

**Reply by prods** [reply]
@ManuelTS  I would love to have the same feature and upvoted this idea.  But I figured out a way of doing this, not straight forward, but it works. Click on the "..." button at the top of the collection and select "Duplicate". Once on the "Duplicate Collection" dialog you should be able to pick the target project. Then you can remove the old instance from the source project. Hope this helps.

**Reply by sebduggan** [reply]
Thanks @prods - that's not too arduous a workaround for some one-time reorganisation. But would still like to see it made easier!

**Comment by subnetmarco** [comment]
I came across this discussion and I am prioritizing the ability to move files (collections, design specs, mocks..) from one project to another, whether it's in the same organization or another.

**Reply by anli-xsigns** [reply]
Super 👍 I would like to move a collection/Project from one workspace to another and found no solution yet.

**Comment by subnetmarco** [comment]
We are aiming at releasing this capability in September in the upcoming Insomnia 10.x

**Comment by cliffordp** [comment]
I'm new to Insomnia and using v10 and during import of Postman Collections (which have crazy file names so I don't know which is which until imported), I then found this thread because I imported all to my one Project and now that I know which is which, I want to move them around into other Projects. tyvm.

**Reply by yaoweiprc** [reply]
You can duplicate a collection to another project and delete the old one manually.

**Comment by subnetmarco** [comment, marked as answer]
By pressing the burger menu (three dots) you can duplicate / move your files to another organization, or project.

![Screenshot 2024-10-15 at 3 00 54 PM](https://github.com/user-attachments/assets/944870f8-8e47-460e-a40b-a5a134dea098)

**Comment by verglor** [comment]
Hi, the duplicate does not contain stored requests/responses so it cannot be used to fully move the collection without data loss. Can we either duplicate these too or have separate move command ?
